compiler/src/main/scala/dx/compiler/WorkflowCompiler.scala

```diff
@@ -167,6 +177,168 @@ case class WorkflowCompiler(extras: Option[Extras],
     (commonMeta ++ workflowMeta, commonDetails)
   }
 
+  private def listMd(s: String): String = {
+    "\n* " + s
+  }
+
+  private def listMd2(s: String): String = {
+    "\n    * `" + s + "`"
+  }
+
+  // Summarize reportable dependencies from details of a workflow
+  private def reportableDependenciesFromWorkflow(
+      workflow: DxExecutable
+  ): Option[String] = {
+    val description = workflow.describe(Set(Field.Name, Field.Details, Field.Hidden)) match {
+      case w: DxWorkflowDescribe => w
+      case _                     => throw new Exception(s"Expected ${workflow.id} to be a workflow.")
+    }
+
+    // Do not generate description section if hidden
+    if (description.hidden.isDefined && description.hidden.get) {
+      None
+    }
+
+    // Generate Markdown description section
+    val headerMd = s"\n## Sub-workflow ${description.name}"
+    val details = description.details.collect {
+      case v: JsValue => v.asJsObject
+    }
+    val md = details match {
+      case Some(JsObject(details)) => {
+        val appsMd = details.get(Constants.NativeAppDependencies) match {
+          case Some(JsArray(a)) => listMd("Native apps") + a.map(v => listMd2(v.toString)).mkString
```

Comment:
use format string

Reply:
Acutally I'd just build up a nested Vector of everything you want to concatenate and then at the very end do a `.flatten.mkString`.

So

```scala
Vector(listMd("Native apps")) ++ a.map(v => listMd2(v.toString))
```

Reply:
Doing flatten.mkString at the end of that would change the logic because you are always putting "Native apps," regardless of whether there is anything to go in the sub-list under it. My goal is to add the header for some type of dependency only when that type of dependency is present.

Reply:
That would still be the case. Maybe you're not understanding the suggestion - might be easier to talk through in person.

Reply:
I think I'm misunderstanding the suggestion -- to discuss in person. @jdidion

Reply:
Another way you can do it:

```
Vector(
  Option.when(nativeAppDependencies.nonEmpty)(Vector(listMd("Native apps")) ++ a.map(v => listMd2(v.toString))),
  Option.when(...)
).flatten.flatten.mkString
```

Reply:
I've used that style in WorkflowCompiler.summarizeReportableDependencies, but elsewhere (where also unwrapping JsValue types into Scala types) I think it's less readable not to use "match," so elsewhere I'm just going to edit for using interpolation instead of concatenation.